app/scripts/services/membership/roles.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .factory('RolesService', function($q, DataService) {
+
+    var getClusterRoles = function() {
+      return DataService.get('clusterroles', null, {});
+    };
+
+    var getRoles = function(context) {
+      return DataService.get('roles', null, context);
+    };
+
+    // return set of clusterRoles & local roles
+    var getAllRoles = function(context) {
+      return $q.all([
+        getRoles(context),
+        getClusterRoles()
+      ]);
+    };
+
+    var listClusterRoles = function(cb, opts) {
+      DataService.list('clusterroles', {}, cb, opts);
+    };
+
+    var listRoles = function(context, cb, opts) {
+      DataService.list('roles', context, cb, opts);
+    };
+
+
+    // wraps the 2 API list calls in a promise for easier use, similar to above get()
+    var listAllRoles = function(context) {
+      var deferred = $q.defer();
+      var result = [];
+
+      var resolve = function(resp) {
+        result.push(resp.by('metadata.name'));
+        if(_.isEqual(result.length, 2)) {
+          deferred.resolve(result);
+        }
+      };
+
+      listRoles(context, function(resp) {
+        resolve(resp);
+      });
+      listClusterRoles(function(resp) {
+        resolve(resp);
+      });
+
+      return deferred.promise;
+    };
+
+
+    return {
+      getRoles: getRoles,
+      getClusterRoles: getClusterRoles,
+      getAllRoles: getAllRoles,
+      listRoles: listRoles,
+      listClusterRoles: listClusterRoles,
+      listAllRoles: listAllRoles
```

Comment:
Again I don't like the wrappers that are just hiding a single call to DataService, I'm ok with things that are hiding the complexity of making multiple requests.  Based on what is in here, the only thing that should stay is listAllRoles